Hemu21/Codelabz
# GitHub Task 1 Completed

Github Task 1 :

- Like/Dislike Feature Implementation:
- Implement database interactions for user likes and dislikes.
- Implement them for tutorials and comments.
- Ensure the feature aligns with the existing database design.

## Description

I created the functions in store folder actions. That functions are used to get the likes data and add or update the likes data. I changed the functionality according to it.  I followed the current database schema. I fixed all necessary files to implement the like/dislike feature for tutorials and comments. I Ensure this feature aligns with the existing database design.

## Related Issue

### Steps to Reproduce Issue:

- Setup the project using main codebase from c2si repository.
- Now login in the website hosted in your localhost.
- Now go to tutorial page and create new tutorial.
- Now go to home page you can find the created tutorial. Click on like and Dislike. You can see change, but you reload page then changed are not updated because it is not stored in fire store database.
- Now add the comment to that tutorial. Click on like and Dislike. You can see change, but you reload page the changed are not updated because it is not stored in fire store database.
- Here the bug is User Unable like/dislike the tutorials and comments.

## Motivation and Context

This change is necessary because of this issue user is unable to like/dislike the tutorials and comments. comments or tutorials without user interaction like/dislike will leads to lack of user experience. by solving this issue user can interact with like/dislike the tutorial and comments with this we can understand which tutorial is most liked by the users. I solved it. Now user can like/dislike the tutorials and comments this will increase user experience.

## How Has This Been Tested?

### Steps to Test:

- Setup the project.
- Now login on the website hosted in your localhost.
- Now go to tutorial page and create new tutorial.
- Now go to home page you can find the created tutorial. Click on like and Dislike. You can see change, if you reload page then changes should update. because it is now stored in fire store database.
- Now add the comment to that tutorial. Click on like and Dislike. You can see change, if you reload page then changes should update because it is now stored in fire store database.
- Now, User able like/dislike the tutorials and comments as shown below.

## Screenshots or GIF (In case of UI changes):

https://github.com/Hemu21/Codelabz/assets/106808387/c707ecec-fdb1-4fe0-b202-4f6b88e2b078

## Types of changes




## Checklist:
